nbdev/showdoc.py

```diff
@@ -243,9 +244,64 @@ def _format_cls_doc(cls, full_name):
     if parent_class != object: args += f' :: {doc_link(get_name(parent_class))}'
     return name,args
 
+# Cell
+def _generate_arg_string(argument_dict):
+    "Turns a dictionary of argument information into a useful docstring"
+    arg_string = '|**Parameters**|Type|Default|'
+    border_string = '|---|---|---|'
+    _has_docment = any(item["docment"] != None for item in argument_dict.values())
+    if _has_docment:
+        arg_string += 'Details|'
+        border_string += '---|'
+    arg_string+= f'\n{border_string}\n'
+    for key, item in argument_dict.items():
+        is_required=True
+        if key == 'return': continue
+        if item['default'] != inspect._empty:
+            is_required = False
+        arg_string += f"|**`{key}`**|"
+        arg_string += "*None Specified*|" if item['anno'] == inspect._empty else f"`{item['anno']}`|"
+        arg_string += "*No Default*|" if is_required else f"`{str(item['default'])}`|"
+        if _has_docment:
+            if item['docment'] is not None:
+                arg_string += f"{item['docment']}|"
+            else:
+                arg_string += "*No Content*|"
+        arg_string += '\n'
+    return arg_string
+
+# Cell
+def _generate_return_string(return_dict:dict):
+    "Turns a dictionary of return information into a useful docstring"
+    return_string = "|**Return Type**|\n|-|\n|"
+    return_string += f"`{return_dict['anno']}`|"
+    if return_dict['docment'] is not None:
+        return_string = return_string.replace("|**Return Type**|", "|**Return Type**|Details|")
+        return_string = return_string.replace("|-|", "|-|-|")
+        return_string += f"{return_dict['docment']}|"
+    return return_string
+
+# Cell
+def _format_args(elt):
+    "Generates a formatted argument string"
+    ment_dict = docments(elt, full=True)
+    ret = None
+    arg_string = ""
+    return_string = ""
+    if "self" in ment_dict.keys(): ment_dict.pop("self")
```

Comment:
None of these conditionals are needed - just add the optional `default` param to `pop`